src/error_display.rs

```diff
@@ -37,8 +37,23 @@ pub fn print_query_error(err: &Error, query: &str, verbose: bool)
     let detail = err.headers().get(&FIELD_DETAILS)
         .and_then(|x| String::from_utf8(x.to_vec()).ok());
     let files = SimpleFile::new("query", query);
+    let context_error = err
+        .contexts()
+        .rev()
+        .map(|s| s.as_ref())
+        .collect::<Vec<_>>();
+    if context_error.len() > 0 {
+        print::error(context_error.join(": "));
+    }
     let diag = Diagnostic::error()
-        .with_message(&format!("{:#}", err))
+        .with_message(&format!(
+            "{:#}{:#}",
+            err.kind_name(),
+            err
+                .initial_message()
+                .map(|s| format!(": {:#}", s))
```

Comment:
nit: since these are strings, you don't need alternate `{:#}` formatting anywhere. Just `{}` is fine. It's not a problem here, but confuse the reader thinking that something special is going on here.